new-log-viewer/src/components/Popups/index.tsx

```diff
@@ -0,0 +1,37 @@
+import {useContext} from "react";
+
+import {
+    Snackbar,
+    Stack,
+} from "@mui/joy";
+
+import {NotificationContext} from "../../contexts/NotificationContextProvider";
+import PopupMessageBox from "./PopupMessageBox";
+
+import "./index.css";
+
+
+/**
+ * Displays popups.
+ *
+ * @return
+ */
+const Popups = () => {
+    const {popupMessages} = useContext(NotificationContext);
+    return (
+        <Snackbar
+            className={"pop-up-messages-container-snackbar"}
+            open={0 < popupMessages.length}
+        >
+            <Stack gap={1}>
+                {popupMessages.map((message, index) => (
```

Comment:
I just realize we shouldn't use `index` as the key for each `PopupMessageBox` here since the array elements can be moving every time a message is removed from the queue.

Reply:
Let's assign a unique id for each message and use it as the key here then. I guess we can also update the message remover to filter on IDs to avoid confusions in future.